packages/mui-base/src/useLink/useLink.ts

```diff
@@ -0,0 +1,163 @@
+'use client';
+import * as React from 'react';
+import {
+  unstable_useForkRef as useForkRef,
+  unstable_useIsFocusVisible as useIsFocusVisible,
+} from '@mui/utils';
+import { UseLinkParameters, UseLinkReturnValue, UseLinkRootSlotProps } from './useLink.types';
+import { extractEventHandlers } from '../utils/extractEventHandlers';
+import { useRootElementName } from '../utils/useRootElementName';
+import { EventHandlers } from '../utils/types';
+import { MuiCancellableEvent } from '../utils/MuiCancellableEvent';
+/**
+ *
+ * API:
+ *
+ * - [useLink API](https://mui.com/base-ui/api/use-link/)
+ */
+export function useLink(parameters: UseLinkParameters = {}): UseLinkReturnValue {
+  const {
+    disabled = false,
+    href,
+    rootRef: externalRef,
+    tabIndex,
+    to,
+    rootElementName: rootElementNameProp = 'a',
+  } = parameters;
+  const linkRef = React.useRef<HTMLLinkElement | HTMLAnchorElement | HTMLElement>();
+
+  const [active, setActive] = React.useState<boolean>(false);
+
+  const [rootElementName, updateRootElementName] = useRootElementName({
+    rootElementName: rootElementNameProp,
+    componentName: 'Link',
+  });
+
+  const isNativeLink = () => {
+    return rootElementName === 'A' && (linkRef.current as HTMLAnchorElement)?.href;
+  };
+
+  const createHandleMouseDown = (otherHandlers: EventHandlers) => (event: React.MouseEvent) => {
+    if (!disabled) {
+      setActive(true);
+      document.addEventListener(
+        'mouseup',
+        () => {
+          setActive(false);
+        },
+        { once: true },
+      );
+    }
+
+    otherHandlers.onMouseDown?.(event);
+  };
+
+  const createHandleKeyDown =
+    (otherHandlers: EventHandlers) => (event: React.KeyboardEvent & MuiCancellableEvent) => {
+      otherHandlers.onKeyDown?.(event);
+
+      if (event.defaultMuiPrevented) {
+        return;
+      }
+
+      if (event.target === event.currentTarget && !isNativeLink() && event.key === ' ') {
+        event.preventDefault();
+      }
+
+      if (event.target === event.currentTarget && event.key === ' ' && !disabled) {
+        setActive(true);
+      }
+
+      // Keyboard accessibility for non interactive elements
```

Comment:
Let's remove the code to support custom DOM elements. A link should be an `a`.

Reply:
Done